data-manipulation/checksum/luhn/validate-credit-card-number-with-luhn-algorithm.yml

```diff
@@ -23,10 +25,15 @@ rule:
           - number: 0x9
         description: Digital root lookup table
       - basic block:
-        - and:
-          - number: 0x30
-          - mnemonic: sub
-          description: Conversion of chr to int 
+        - or:
+          - and:
+            - number: 0x30
+            - mnemonic: sub
+            description: Conversion of chr to int (SUB 0x30) 
+          - and:
+            - mnemonic: lea
+            - offset: -0x30
```

Comment:
i don't think negative numbers are supported.

right @mike-hunhoff ? or will this work for offsets, just not numbers?

Reply:
**Short answer:** no we do not officially support negative numbers

**Because**: By default both the viv and IDA APIs default to two's complement for representing numbers so capa can't match `-1` because it only sees the two's complement form `0xFFFFFFFF`.

**However:** It appears that the viv API does not default to two's complement for representing offsets which allows `0x-30` to match. 

Unfortunately the IDA API does default to two's complement for representing offsets therefore, and I haven't tested this yet but I assume, this rule will match in viv but not in IDA.

I would probably consider this a bug as we need to establish a consistent representation of numbers, offsets, etc. across capa and the extractors.

Offsets in viv:
```
('Offset', -4)
('Offset', -20)
('Offset', -20)
('Offset', 0)
('Offset', -20)
('Offset', -4)
('Offset', 16)
('Offset', 0)
('Offset', 20)
('Offset', -4)
```

Offsets in IDA:
```
Python>insn = sark.code.instruction.Instruction(here())
Python>insn.operands[1].offset
0xfffffef8
```